socket.cpp

```diff
@@ -56,7 +56,10 @@ bool plain_socket::set_hostname(const char* sAddr)
 	sAddrMb[ln] = '\0';
 
 	if ((sTmp = strstr(sAddrMb, "//")) != nullptr)
+	{
+		sTmp += 2;
```

Comment:
This can produce a buffer overflow if the user add ony `stratum+tcp//` as address, can you please add a check that the new pointer is not behind '/0'